File: Kirameku-backend/app/schemas/download.py
from datetime import datetime

from pydantic import BaseModel


class DownloadFileCreate(BaseModel):
    title: str
    description: str = ""
    category: str = ""
    source_type: str = "link"
    external_url: str
    original_filename: str = ""
    file_size: int = 0
    mime_type: str = ""
    is_public: bool = True
    sort: int = 0


class DownloadFileUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    category: str | None = None
    source_type: str | None = None
    external_url: str | None = None
    original_filename: str | None = None
    file_size: int | None = None
    mime_type: str | None = None
    is_public: bool | None = None
    sort: int | None = None


class DownloadFileOut(BaseModel):
    id: int
    title: str
    description: str
    category: str
    source_type: str
    original_filename: str
    external_url: str
    file_size: int
    mime_type: str
    is_public: bool
    download_count: int
    sort: int
    download_url: str
    created_at: datetime
    updated_at: datetime | None = None

File: Kirameku-backend/admin/src/api/download.ts
import { http } from "@/utils/http";

export type DownloadSource = "local" | "cloudreve" | "onedrive" | "link";

export type DownloadFileItem = {
  id: number;
  title: string;
  description: string;
  category: string;
  source_type: DownloadSource;
  original_filename: string;
  external_url: string;
  file_size: number;
  mime_type: string;
  is_public: boolean;
  download_count: number;
  sort: number;
  download_url: string;
  created_at: string;
  updated_at: string | null;
};

export const getDownloadFiles = (params?: {
  kind?: "all" | "files" | "music";
}) => {
  return http.request<DownloadFileItem[]>("get", "/api/downloads/admin", {
    params
  });
};

export const createExternalDownload = (
  data: Partial<DownloadFileItem> & { external_url: string }
) => {
  return http.request<DownloadFileItem>("post", "/api/downloads", { data });
};

export const updateDownloadFile = (
  fileId: number,
  data: Partial<DownloadFileItem>
) => {
  return http.request<DownloadFileItem>("put", `/api/downloads/${fileId}`, {
    data
  });
};

export const deleteDownloadFile = (fileId: number) => {
  return http.request<{ ok: boolean }>("delete", `/api/downloads/${fileId}`);
};

export const uploadDownloadFile = (data: FormData) => {
  return http.request<DownloadFileItem>("post", "/api/downloads/upload", {
    data,
    headers: { "Content-Type": "multipart/form-data" }
  });
};
